### PR TITLE
feature/add-header-to-the-ssspace-project

### PR DESCRIPTION
**Pull request by Aleksandra05sh**
Add Header to the SSSpace project (https://trello.com/c/8a63HHpr/2-add-header-to-the-ssspace-project)
image

- added a logo 
- added navigation: home, events, agencies, astronauts, about
- added icons for socials: insta(NASA), github, LinkedIn(SoftServe:

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+import theme from './theme';
 
 function App() {
-  return <div>Hello</div>;
+  const elements = useRoutes(rootRoutes);
+  return (
+    <ThemeProvider theme={theme}>
+      <>
+        <Header></Header>
+        <div style={{ marginTop: '64px' }}></div>
```
Comment (Aleksandra05sh) [inline src/App.tsx:14]: just for now <div> with style  to see information on Pages

File: src/components/Header/Header.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+.navPage {
+  margin: 4px;
+}
+
+.socialNetworks {
+  display: flex;
+  justify-content: flex-end;
+}
+.socialNetworks > .socialNetwork {
```
Comment (Aleksandra05sh) [inline src/components/Header/Header.scss:23]: add two classes to increase the weight of selectors (using one class the styles didn't work)

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  { component: GitHubIcon, url: gitHubUrl },
+];
+
+function Header() {
+  return (
+    <AppBar position="fixed">
+      <Container maxWidth="xl">
+        <Toolbar>
+          <Box className="logo" component={Link} to={homeUrl}>
+            <img src={process.env.PUBLIC_URL + '/logo.png'} alt="Logo" />
```
Comment (Aleksandra05sh) [inline src/components/Header/Header.tsx:37]: just '/logo.png' doesn't work

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -1,3 +1,63 @@
...
+                key={url}
+                variant="text"
+                size="large"
+                to={url}
+                component={Link}
+              >
+                {name}
+              </Button>
+            ))}
+          </Box>
```
Comment (kerrimov) [inline src/components/Header/Header.tsx:46]: Create 2 components:  NavigationList and move logic here

**Review by kerrimov — CHANGES_REQUESTED** [review]
Please do not forget to provide types

File: src/theme.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { createTheme, responsiveFontSizes } from '@mui/material/styles';
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/theme.ts:1]: done

File: src/components/Header/socials.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const instagramUrl = 'http://instagram.com/nasa';
+export const linkedInUrl = 'https://www.linkedin.com/company/softserve/';
+export const gitHubUrl = 'https://github.com/kerrimov/the-ssspace';
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Header/socials.ts:3]: done

File: src/components/Header/Header.tsx [hunk 7]
```diff
@@ -1,3 +1,63 @@
 import React from 'react';
+import { Link } from 'react-router-dom';
+import { AppBar, Box, Toolbar, Container, Button } from '@mui/material';
+import InstagramIcon from '@mui/icons-material/Instagram';
+import LinkedInIcon from '@mui/icons-material/LinkedIn';
+import GitHubIcon from '@mui/icons-material/GitHub';
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Header/Header.tsx:6]: 1. done
2.

File: src/components/Header/Header.tsx [hunk 8]
```diff
@@ -1,3 +1,63 @@
...
+  { name: 'Agencies', url: RoutesPath.AGENCIES },
+  { name: 'Astronauts', url: RoutesPath.ASTRONAUTS },
+  { name: 'About', url: RoutesPath.ABOUT },
+];
+
+const socials = [
+  { component: InstagramIcon, url: instagramUrl },
+  { component: LinkedInIcon, url: linkedInUrl },
+  { component: GitHubIcon, url: gitHubUrl },
+];
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Header/Header.tsx:23]: 1. done
2. done

File: src/App.tsx [hunk 9]
```diff
@@ -4,15 +4,14 @@ import { Routes } from './components/Routes';
...
   return (
-    <>
-      <ThemeProvider theme={theme}>
-        <Header />
-        {routes}
-      </ThemeProvider>
-    </>
+    <ThemeProvider theme={theme}>
+      <Header></Header>
+      <div className="root-wrapper">{routes}</div>
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/App.tsx:14]: done

File: src/App.scss [hunk 10]
```diff
@@ -0,0 +1,3 @@
+.root-wrapper {
+  margin-top: 4rem;
+}
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/App.scss:3]: done

File: src/components/Header/Header.tsx (hunk 4 above)
Comment (Aleksandra05sh) [reply src/components/Header/Header.tsx:46]: NavigationList and SocialsList were created

File: src/components/Header/Header.tsx [hunk 11]
```diff
@@ -1,3 +1,63 @@
...
+
+export const Header = () => {
+  return (
+    <AppBar className="header" position="fixed">
+      <Container maxWidth="xl">
+        <Toolbar>
+          <Box className="header-logo" component={Link} to={RoutesPath.HOME}>
+            <img src={process.env.PUBLIC_URL + '/logo.png'} alt="Logo" />
+          </Box>
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Header/Header.tsx:32]: done

File: src/components/Header/Header.tsx [hunk 12]
```diff
@@ -1,3 +1,63 @@
...
+              <Button
+                className="header-socials-item"
+                key={url}
+                href={url}
+                target="_blank"
+              >
+                <Component />
+              </Button>
+            ))}
+          </Box>
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Header/Header.tsx:58]: 1.  SocialsList was created

File: src/components/Header/components/Logo/Logo.scss [hunk 13]
```diff
@@ -0,0 +1,7 @@
+.header-logo {
+  display: flex;
+  justify-content: flex-start;
+  width: 3.125rem;
+  height: 3.125rem;
+  margin-right: 1.5re;
```
Comment (kerrimov) [inline src/components/Header/components/Logo/Logo.scss:6]: `rem`

File: src/components/Header/components/NavigationList/NavigationList.tsx [hunk 14]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Box, Button } from '@mui/material';
+import { pages } from '../../constants';
+import './NavigationList.scss';
+
+export const NavigationList = () => (
+  <Box className="header-navigation" component="nav">
+    {pages.map(({ name, url }) => (
```
Comment (kerrimov) [inline src/components/Header/components/NavigationList/NavigationList.tsx:9]: add `id` for list items

File: src/components/Header/socialsUrl.ts [hunk 15]
```diff
@@ -0,0 +1,5 @@
+export const enum SocialsUrl {
```
Comment (kerrimov) [inline src/components/Header/socialsUrl.ts:1]: move this enum to `socials.ts` and remove this file

File: src/components/Header/constants/index.ts [hunk 16]
```diff
@@ -0,0 +1,2 @@
+export { pages } from './pages';
```
Comment (kerrimov) [inline src/components/Header/constants/index.ts:1]: index file could be deleted

File: src/App.tsx [hunk 17]
```diff
@@ -1,18 +1,17 @@
 import React from 'react';
 import { useRoutes } from 'react-router-dom';
 import { Routes } from './components/Routes';
+import { Router } from './components/Router';
 import { Header } from './components/Header';
 import { ThemeProvider } from '@mui/material';
 import { theme } from './theme/theme';
 
 export const App = () => {
   const routes = useRoutes(Routes);
```
Comment (kerrimov) [inline src/App.tsx:10]: move to Router.tsx

**Review by kerrimov — CHANGES_REQUESTED** [review]
Delete App.scss if it is not needed

File: src/components/Header/components/Logo/Logo.scss (hunk 13 above)
Comment (Aleksandra05sh) [reply src/components/Header/components/Logo/Logo.scss:6]: done

File: src/components/Header/components/NavigationList/NavigationList.tsx (hunk 14 above)
Comment (Aleksandra05sh) [reply src/components/Header/components/NavigationList/NavigationList.tsx:9]: done

File: src/components/Header/constants/index.ts (hunk 16 above)
Comment (Aleksandra05sh) [reply src/components/Header/constants/index.ts:1]: done

**Review by kerrimov — COMMENTED** [review]
rename file `Routes.tsx` to `routes.tsx` and const Routes too

File: src/components/Header/components/SocialsList/SocialsList.tsx [hunk 18]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Box, Button } from '@mui/material';
+import { socials } from '../../constants/socials';
+import './SocialsList.scss';
+
+export const SocialsList = () => (
+  <Box className="header-socials">
+    {socials.map(({ icon: Icon, url }, index) => (
```
Comment (kerrimov) [inline src/components/Header/components/SocialsList/SocialsList.tsx:8]: type should be provided

File: src/components/Header/components/NavigationList/NavigationList.tsx [hunk 19]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { Box, Button } from '@mui/material';
+import { pages } from '../../constants/pages';
+import './NavigationList.scss';
+
+export const NavigationList = () => (
+  <Box className="header-navigation" component="nav">
+    {pages.map(({ name, url }, index) => (
```
Comment (kerrimov) [inline src/components/Header/components/NavigationList/NavigationList.tsx:9]: types should be provided

File: src/components/Header/constants/socials.ts [hunk 20]
```diff
@@ -0,0 +1,13 @@
+import { Instagram, LinkedIn, GitHub } from '@mui/icons-material';
+
+export const enum SocialsUrl {
+  INSTAGRAM = 'http://instagram.com/nasa',
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Header/constants/socials.ts:4]: done

File: src/components/Header/components/NavigationList/NavigationList.tsx (hunk 19 above)
Comment (Aleksandra05sh) [reply src/components/Header/components/NavigationList/NavigationList.tsx:9]: 1. created interface Page in pages.ts
2. added type for index

File: src/components/Router/Router.tsx [hunk 21]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import './Router.scss';
+import { useRoutes } from 'react-router-dom';
+import { Routes } from './Routes';
+
+export const Router = () => {
+  const routes = useRoutes(Routes);
```
[the remark this answers is not in the document]
Comment (Aleksandra05sh) [reply src/components/Router/Router.tsx:7]: done

File: src/components/Header/constants/pages.ts [hunk 22]
```diff
@@ -0,0 +1,10 @@
+import { RoutesPath } from '../../Router/routesPath';
+import { RoutesName, Page } from '../types/pages.type';
```
Comment (kerrimov) [inline src/components/Header/constants/pages.ts:2]: use `import type` for importing types

**Review by kerrimov — COMMENTED** [review]
rename files in the folder `types` to `filename.ts`

File: src/components/Header/constants/pages.ts (hunk 22 above)
Comment (Aleksandra05sh) [reply src/components/Header/constants/pages.ts:2]: but in this case, we will have two files with the same name and it will be not convenient

**Review by kerrimov — APPROVED** [review]

**Pull request merged**